### DISCUSSION TITLE
Visualization with CanvasGrid - AttributeError: 'MyModel' object has no attribute 'grid'

### DISCUSSION THREAD

**Discussion by romain-sen** [mesa/mesa#2102] [Q&A]
Hello, 
I am having trouble to visualize my model I created. I did all the steps I did with the MoneyModel example and yet I am still having an error when I try the visualization with CanvasGrid : `AttributeError: 'NuclearWasteModel' object has no attribute 'grid'`

I initialize my grid in my model so why am I having this error ?

please find my code here : https://github.com/romain-sen/MAS-nuclear-wates/tree/feat/visualization

**Comment by rht** [comment]
Because this class import, https://github.com/romain-sen/MAS-nuclear-wates/blob/1d681eb1ac900d8f5a1a620c24e602d785e37de3/without-communication/visualization.py#L9, overshadows the import in L1.

**Reply by romain-sen** [reply]
Thank you very much ! I haven't mastered the types in python very well yet
